src/geom/elem.C

```diff
@@ -3404,4 +3404,37 @@ Elem::is_internal(const unsigned int i) const
   }
 }
 
+bool
+Elem::edge_orientation(const unsigned int i) const
+{
+  libmesh_assert_less (i, this->n_edges());
+
+  return this->point(this->local_edge_node(i, 0)) >
+         this->point(this->local_edge_node(i, 1));
+}
+
+bool
+Elem::face_orientation(const unsigned int i) const
+{
+  libmesh_assert_less (i, this->n_faces());
+
+  const unsigned int N = Elem::type_to_n_sides_map[this->side_type(i)];
+
+  const std::vector<unsigned int> nodes = this->nodes_on_side(i);
+  std::vector<Point> vertices(N);
```

Comment:
This sequence of lines confused me because it looks like `N` should be a number of sides, since it comes from the `type_to_n_sides_map`, but you are getting a number of vertices, based on the fact that "the number of sides of a side" is equal to the number of vertices on that side for 3D elements. My suggestions are to name the variable more descriptively, e.g. `n_side_vertices`, and even better, put a comment explaining what you are doing.